packages/docs/api/equality-algorithm.md

```diff
@@ -0,0 +1,225 @@
+---
+id: equality-algorithm
+title: Equality Algorithm
+---
+
+The equality algorithm is an extensible and parameterized deep equality checker.
+It is used by all functions that check deep equality inside earl.
+
+## Algorithm
```

Comment:
What do you think about linking to source code of functions implementing the algorithm here?

Reply:
I don't like linking to code from the docs because it gets out of sync to easily. I think linking to the docs from the code is enough.